libff/algebra/curves/tests/test_groups.cpp

```diff
@@ -166,19 +187,61 @@ void test_group_membership_valid()
 }
 
 template<typename GroupT>
-void test_group_membership_invalid_g1(const typename GroupT::base_field &x)
+void test_group_membership_proof_valid()
 {
-    const typename GroupT::base_field x_squared = x * x;
-    const typename GroupT::base_field x_cubed = x_squared * x;
-    const typename GroupT::base_field y_squared =
-        x_cubed + (GroupT::coeff_a * x_squared) + GroupT::coeff_b;
-    const typename GroupT::base_field y = y_squared.sqrt();
-    const GroupT g1_invalid(x, y, GroupT::base_field::one());
+    for (size_t i = 0 ; i < 1000 ; ++i)
+    {
+        const GroupT g = GroupT::random_element();
+        const GroupT membership_proof = g.proof_of_safe_subgroup();
+        assert(membership_proof.mul_by_cofactor() == g);
+    }
+}
 
+template<typename GroupT>
+bool profile_group_membership()
```

Comment:
What about moving that to a `profiling` folder? As you wrote, this is not testing correctness here.

Reply:
Definitely makes sense to separate these.  I was thinking of doing it separately, as it involves another kind of executable and new make targets etc.

Reply:
(Just noticed that the profile target and executable already exist, so I'll do this as part of this PR)

Reply:
This should be pushed now.